modules/core/src/v2/coinFactory.ts

```diff
@@ -126,11 +127,16 @@ GlobalCoinFactory.registerCoinConstructor('susd', Susd.createInstance);
 GlobalCoinFactory.registerCoinConstructor('tsusd', Tsusd.createInstance);
 
 for (const token of [...tokens.bitcoin.eth.tokens, ...tokens.testnet.eth.tokens]) {
-  const tokenConstructor = Token.createTokenConstructor(token);
+  const tokenConstructor = Erc20Token.createTokenConstructor(token);
   GlobalCoinFactory.registerCoinConstructor(token.type, tokenConstructor);
   GlobalCoinFactory.registerCoinConstructor(token.tokenContractAddress, tokenConstructor);
 }
 
+for (const token of [...tokens.bitcoin.xlm.tokens, ...tokens.testnet.xlm.tokens]) {
```

Comment:
Any idea why we construct both mainnet and testnet tokens in every env? I know you just copied the style above

Reply:
I think it's because coinFactory doesn't have the concept of environment, it just receives a string and tries to get an instance of the coin.